school_attendance/models/school_attendance.py

```diff
@@ -0,0 +1,1204 @@
+            if attendance_sheet_data.one_9:
+                att_count = att_count + 1
+            if attendance_sheet_data.one_0:
+                att_count = att_count + 1
+
+            if attendance_sheet_data.two_1:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_2:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_3:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_4:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_5:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_6:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_7:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_8:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_9:
+                att_count = att_count + 1
+            if attendance_sheet_data.two_0:
+                att_count = att_count + 1
+            if attendance_sheet_data.three_1:
+                att_count = att_count + 1
+            percentage = (float(att_count / 31.00)) * 100
+            attendance_sheet_data.percentage = percentage
+        return res
+
+    _description = "Attendance Sheet Line"
+    _name = "attendance.sheet.line"
+    _order = "roll_no"
+
+    roll_no = fields.Integer(
+        "Roll Number", required=True, help="Roll Number of Student"
+    )
+    standard_id = fields.Many2one(
+        "attendance.sheet",
+        "Standard",
+        help="Attendance for particular standard",
+    )
+    name = fields.Char(
+        "Student Name", required=True, readonly=True, help="Enter student name"
+    )
+    one = fields.Boolean("1")
+    two = fields.Boolean("2")
+    three = fields.Boolean("3")
+    four = fields.Boolean("4")
+    five = fields.Boolean("5")
+    seven = fields.Boolean("7")
+    six = fields.Boolean("6")
+    eight = fields.Boolean("8")
+    nine = fields.Boolean("9")
+    ten = fields.Boolean("10")
+    one_1 = fields.Boolean("11")
+    one_2 = fields.Boolean("12")
+    one_3 = fields.Boolean("13")
+    one_4 = fields.Boolean("14")
+    one_5 = fields.Boolean("15")
+    one_6 = fields.Boolean("16")
+    one_7 = fields.Boolean("17")
+    one_8 = fields.Boolean("18")
+    one_9 = fields.Boolean("19")
+    one_0 = fields.Boolean("20")
+    two_1 = fields.Boolean("21")
+    two_2 = fields.Boolean("22")
+    two_3 = fields.Boolean("23")
+    two_4 = fields.Boolean("24")
+    two_5 = fields.Boolean("25")
+    two_6 = fields.Boolean("26")
+    two_7 = fields.Boolean("27")
+    two_8 = fields.Boolean("28")
+    two_9 = fields.Boolean("29")
+    two_0 = fields.Boolean("30")
+    three_1 = fields.Boolean("31")
+    percentage = fields.Float(
+        compute="_compute_percentage",
+        method=True,
+        string="Attendance (%)",
+        store=False,
+        help="Percentage",
+    )
```

Comment:
Timesheet has a good widget to solve this thing.
Analyse guys.
@AnuKPatel we have this since long, we should make smart code, we cannot continue.
Its painful if we do not use parent modules.